apps/test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx

```diff
@@ -311,6 +311,26 @@ describe('useSchoolInfo', () => {
         expect(schoolNameSessionStorageCalls[1][1]).toBe('Super Cool School');
       });
     });
+
+    it('should reset school info to initial state if passed', async () => {
```

Comment:
@drizco this test seemed to cause other tests to fail if I put it higher up in the file. Is that a known issue? Or should we file a ticket to investigate?

Reply:
hmm, yeah that sounds sketchy... I'm happy to investigate this

Reply:
Filed a [ticket](https://codedotorg.atlassian.net/browse/ACQ-2558)!

Here's some test output (in this case, I put this test inside a `describe` block to see if that helped. It didnt so I didnt use one in the end)
```
 FAIL  test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx (19.123 s)
  useSchoolInfo
    ✓ should use initialState instead of sessionStorage if passed (28 ms)
    ✓ should use sessionStorage if no initialState is passed (3 ms)
    hook state updates
      ✕ should send an analytics event (7 ms)
      ✕ should fetch schools (4 ms)
      country updates
        ✓ should update sessionStorage (16 ms)
        ✓ should retain schoolZip, schoolId, and schoolName on country changes (7 ms)
        ✓ should send an analytics event (3 ms)
      reset to initial state
        ✓ should reset school info to initial state if passed (43 ms)
      schoolZip updates
        ✕ should update sessionStorage if zip is valid (7 ms)
        ✕ should clear sessionStorage schoolZip if zip is invalid (4 ms)
        ✓ should retain schoolId and schoolName if the schoolZip changes (11 ms)
      schoolId updates
        ✕ should update sessionStorage (3 ms)
        ✕ should send analytics events (3 ms)
      schoolName updates
        ✕ should update sessionStorage (5 ms)
    fetchSchools
      ✓ returns the school list sorted alphabetically (2 ms)

  ● useSchoolInfo › hook state updates › schoolZip updates › should update sessionStorage if zip is valid

    expect(received).toHaveLength(expected)

    Expected length: 2
    Received length: 1
    Received array:  [["schoolZip", "00000"]]

      202 |             ([key]) => key === SCHOOL_ZIP_SESSION_KEY
      203 |           );
    > 204 |         expect(schoolZipSessionStorageCalls).toHaveLength(2);
          |                                              ^
      205 |         expect(schoolZipSessionStorageCalls[0][1]).toBe(initialState.schoolZip);
      206 |         expect(schoolZipSessionStorageCalls[1][1]).toBe('90210');
      207 |       });

      at toHaveLength (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:204:46)
      at call (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at Generator.tryCatch (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at Generator._invoke [as next] (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at asyncGeneratorStep (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at asyncGeneratorStep (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)

  ● useSchoolInfo › hook state updates › schoolZip updates › should clear sessionStorage schoolZip if zip is invalid

    expect(received).toHaveLength(expected)

    Expected length: 2
    Received length: 1
    Received array:  [["schoolZip", "00000"]]

      218 |             ([key]) => key === SCHOOL_ZIP_SESSION_KEY
      219 |           );
    > 220 |         expect(schoolZipSessionStorageCalls).toHaveLength(2);
          |                                              ^
      221 |         expect(schoolZipSessionStorageCalls[0][1]).toBe(initialState.schoolZip);
      222 |         expect(schoolZipSessionStorageCalls[1][1]).toBe('');
      223 |       });

      at Object.toHaveLength (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:220:46)

  ● useSchoolInfo › hook state updates › should send an analytics event

    expect(jest.fn()).toHaveBeenCalledWith(...expected)

    - Expected
    + Received

      "Valid Zip Code Entered",
      Object {
    -   "zip": "90210",
    +   "zip": "00000",
      },
      "Both",

    Number of calls: 1

      250 |       });
      251 |
    > 252 |       expect(sendAnalyticsEventSpy).toHaveBeenCalledWith(
          |                                     ^
      253 |         EVENTS.ZIP_CODE_ENTERED,
      254 |         {zip: '90210'},
      255 |         PLATFORMS.BOTH

      at toHaveBeenCalledWith (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:252:37)
      at call (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at Generator.tryCatch (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at Generator._invoke [as next] (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at asyncGeneratorStep (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at asyncGeneratorStep (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)

  ● useSchoolInfo › hook state updates › should fetch schools

    expect(jest.fn()).toHaveBeenCalledTimes(expected)

    Expected number of calls: 2
    Received number of calls: 1

      262 |       });
      263 |
    > 264 |       expect(mockFetch).toHaveBeenCalledTimes(2);
          |                         ^
      265 |       expect(mockFetch.mock.calls[0][0]).toEqual(
      266 |         expect.stringMatching(initialState.schoolZip)
      267 |       );

      at toHaveBeenCalledTimes (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:264:25)
      at call (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at Generator.tryCatch (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at Generator._invoke [as next] (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at asyncGeneratorStep (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at asyncGeneratorStep (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)

  ● useSchoolInfo › hook state updates › schoolId updates › should update sessionStorage

    expect(received).toHaveLength(expected)

    Expected length: 2
    Received length: 1
    Received array:  [["schoolId", "1"]]

      281 |             ([key]) => key === SCHOOL_ID_SESSION_KEY
      282 |           );
    > 283 |         expect(schoolIdSessionStorageCalls).toHaveLength(2);
          |                                             ^
      284 |         expect(schoolIdSessionStorageCalls[0][1]).toBe(initialState.schoolId);
      285 |         expect(schoolIdSessionStorageCalls[1][1]).toBe('ABC123');
      286 |       });

      at Object.toHaveLength (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:283:45)

  ● useSchoolInfo › hook state updates › schoolId updates › should send analytics events

    expect(jest.fn()).toHaveBeenCalledWith(...expected)

    - Expected
    + Received

    - "User Clicks I Do Not Teach In A School Setting",
    + "Valid Zip Code Entered",
    - Object {}
    + Object {
    +   "zip": "00000",
    + },
      "Both",

    Number of calls: 1

      291 |         });
      292 |
    > 293 |         expect(sendAnalyticsEventSpy).toHaveBeenCalledWith(
          |                                       ^
      294 |           EVENTS.DO_NOT_TEACH_AT_SCHOOL_CLICKED,
      295 |           {},
      296 |           PLATFORMS.BOTH

      at Object.toHaveBeenCalledWith (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:293:39)

  ● useSchoolInfo › hook state updates › schoolName updates › should update sessionStorage

    expect(received).toHaveLength(expected)

    Expected length: 2
    Received length: 1
    Received array:  [["schoolName", "Cool School"]]

      329 |             ([key]) => key === SCHOOL_NAME_SESSION_KEY
      330 |           );
    > 331 |         expect(schoolNameSessionStorageCalls).toHaveLength(2);
          |                                               ^
      332 |         expect(schoolNameSessionStorageCalls[0][1]).toBe(
      333 |           initialState.schoolName
      334 |         );

      at toHaveLength (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:331:47)
      at call (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at Generator.tryCatch (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at Generator._invoke [as next] (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at asyncGeneratorStep (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at asyncGeneratorStep (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at _next (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)
      at Object.<anonymous> (test/unit/schoolInfo/hooks/useSchoolInfoTest.jsx:2:1)

Test Suites: 1 failed, 1 total
Tests:       7 failed, 8 passed, 15 total
Snapshots:   0 total
Time:        20.227 s
Ran all test suites matching /test\/unit\/schoolInfo\/hooks\/useSchoolInfoTest.jsx/i.
```